apidef/graphql_config_adapter.go

```diff
@@ -0,0 +1,280 @@
+		return nil, err
+	}
+
+	conf.SetFieldConfigurations(fieldConfigs)
+	conf.SetDataSources(datsSources)
+
+	return &conf, nil
+}
+
+func (g *GraphQLConfigAdapter) engineConfigV2FieldConfigs() (planFieldConfigs plan.FieldConfigurations) {
+	for _, fc := range g.config.Engine.FieldConfigs {
+		planFieldConfig := plan.FieldConfiguration{
+			TypeName:              fc.TypeName,
+			FieldName:             fc.FieldName,
+			DisableDefaultMapping: fc.DisableDefaultMapping,
+			Path:                  fc.Path,
+		}
+
+		planFieldConfigs = append(planFieldConfigs, planFieldConfig)
+	}
+
+	return planFieldConfigs
+}
+
+func (g *GraphQLConfigAdapter) engineConfigV2DataSources() (planDataSources []plan.DataSourceConfiguration, err error) {
+	for _, ds := range g.config.Engine.DataSources {
+		planDataSource := plan.DataSourceConfiguration{
+			RootNodes: []plan.TypeField{},
+		}
+
+		for _, typeField := range ds.RootFields {
+			planTypeField := plan.TypeField{
+				TypeName:   typeField.Type,
+				FieldNames: typeField.Fields,
+			}
+
+			planDataSource.RootNodes = append(planDataSource.RootNodes, planTypeField)
+		}
+
+		switch ds.Kind {
+		case GraphQLEngineDataSourceKindREST:
+			restConfig := GraphQLEngineDataSourceConfigREST{}
+			err = json.Unmarshal(ds.Config, &restConfig)
+			if err != nil {
+				return nil, err
+			}
+
+			factory := &restDataSource.Factory{}
+			if g.httpClient != nil {
+				factory.Client = httpclient.NewNetHttpClient(g.httpClient)
+			}
+			planDataSource.Factory = factory
+
+			planDataSource.Custom = restDataSource.ConfigJSON(restDataSource.Configuration{
+				Fetch: restDataSource.FetchConfiguration{
+					URL:    restConfig.URL,
+					Method: restConfig.Method,
+					Body:   restConfig.Body,
+					Query:  g.convertURLQueriesToEngineV2Queries(restConfig.Query),
+					Header: g.convertHttpHeadersToEngineV2Headers(restConfig.Headers),
+				},
+			})
+
+		case GraphQLEngineDataSourceKindGraphQL:
+
+			graphqlConfig := GraphQLEngineDataSourceConfigGraphQL{}
+			err = json.Unmarshal(ds.Config, &graphqlConfig)
+			if err != nil {
+				return nil, err
+			}
+
+			factory := &graphqlDataSource.Factory{}
+			if g.httpClient != nil {
+				factory.Client = httpclient.NewNetHttpClient(g.httpClient)
+			}
+			planDataSource.Factory = factory
+
+			planDataSource.Custom = graphqlDataSource.ConfigJson(graphqlDataSource.Configuration{
+				Fetch: graphqlDataSource.FetchConfiguration{
+					URL:    graphqlConfig.URL,
+					Method: graphqlConfig.Method,
+					Header: g.convertHttpHeadersToEngineV2Headers(graphqlConfig.Headers),
+				},
+			})
+		}
+
+		planDataSources = append(planDataSources, planDataSource)
+	}
+
+	err = g.determineChildNodes(&planDataSources)
+	return planDataSources, err
+}
+
+func (g *GraphQLConfigAdapter) SetHttpClient(httpClient *http.Client) {
+	g.httpClient = httpClient
+}
+
+func (g *GraphQLConfigAdapter) convertURLQueriesToEngineV2Queries(apiDefQueries map[string]string) []restDataSource.QueryConfiguration {
```

Comment:
Is it required to have `convertURLQueriesToEngineV2Queries` and `convertHttpHeadersToEngineV2Headers` helpers as method of adapter? Could it be just helper functions?

Reply:
As it makes only sense that they exist when the `GraphQLAdapter`exists, I think it is better to couple them with the adapter. If you delete the adapter, you also delete the helpers then. Makes sense?